staging/src/k8s.io/apiserver/pkg/util/configmetrics/info_collector.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package configmetrics
+
+import (
+	"sync/atomic"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/utils/ptr"
+)
+
+// HashProvider is an interface for getting the current config hash values
+type HashProvider interface {
+	GetCurrentHashes() map[string]string
+}
+
+// AtomicHashProvider implements HashProvider using atomic operations for thread-safe access
+type AtomicHashProvider struct {
+	hashes map[string]*atomic.Pointer[string]
+}
+
+// NewAtomicHashProvider creates a new atomic hash provider with the specified label names
+func NewAtomicHashProvider(labelNames ...string) *AtomicHashProvider {
```

Comment:
Store label names in this struct so remember its order and can give it to `configInfoCustomCollector`.